src/tests/Microsoft.VisualStudio.Composition.Tests/XunitExtensions/MefFactDiscoverer.cs

```diff
@@ -70,6 +70,11 @@ public MefFactTestCase(IMessageSink diagnosticMessageSink, TestMethodDisplay def
                 this.compositionVersions = factAttribute.CompositionVersions;
                 this.noCompatGoal = factAttribute.NoCompatGoal;
                 this.invalidConfiguration = factAttribute.InvalidConfiguration;
+
+                if (factAttribute.SkipOnMono && Type.GetType("Mono.Runtime") != null)
```

Comment:
Does this code have visibility into the Traits on the test such that we can use that skip instead of a new property on the attribute?